src/controllers/controller.bar.js

```diff
@@ -328,18 +328,13 @@ module.exports = function(Chart) {
 			var me = this;
 			var chart = me.chart;
 			var rects = me.getMeta().data;
-			var dataset = me.getDataset();
 			var ilen = rects.length;
 			var i = 0;
-			var d;
 
 			helpers.canvas.clipArea(chart.ctx, chart.chartArea);
 
 			for (; i < ilen; ++i) {
-				d = dataset.data[i];
-				if (d !== null && d !== undefined && !isNaN(d)) {
```

Comment:
Yes, I like `scale.getRightValue` much better than the old check. I've updated it to that